src/backend/dt.c

```diff
@@ -300,6 +300,98 @@ dt_t **dtdtoff(dt_t **pdtend, dt_t *dt, unsigned offset)
     return dtxoff(pdtend, s, offset);
 }
 
+/**************************************
+ * Repeat a list of dt_t's len times.
+ */
+dt_t **dtrepeat(dt_t **pdtend, dt_t *dt, size_t len)
```

Comment:
Can you call the parameter `num` or something instead of `len`?  I know you're using it for static arrays, as this function isn't specific to them I think `len` isn't a great fit.